### DISCUSSION TITLE
False Classification with unknown language M

### DISCUSSION THREAD

**Discussion by Kiyotoko** [github-linguist/linguist#6532] [Classification] [answered]
Hello everyone,

I use Matlab files in my [repository](https://github.com/Kiyotoko/vault/tree/master) which end with ``.m``. However, some files are not recognized as Matlab, but as M. When I browse the repository, it does not show me which files they are.

Thank you all for your time and support.

**Comment by lildude** [comment, marked as answer]
There is only one file that has been incorrectly identified:

```console
➜ github-linguist vault/ --breakdown
46.96%  6113       Python
38.88%  5062       MATLAB
10.58%  1377       M
3.58%   466        Ruby

[...]

M:
Klasse 11/Physik/Projects/Octave/task_11.m

[...]
➜ 
```

It hasn't been correctly identified because the content didn't match the Matlab regex used to differentiate all the users of the `.m` extension:

https://github.com/github-linguist/linguist/blob/916bd8f3df802fa98b0ea85539e67bd0b88ef158/lib/linguist/heuristics.yml#L400-L417

... so it's fallen through to the classifier which can only guess based on the samples in this repo.

Now things get interesting... the reason the Search that happens when you click the M in the sidebar can't find it appears to be down to a caching issue somewhere. At some point the file was identified as Objective-C (for the same reason as before) and this has been cached as we can see if we [search for the filename instead of the language](https://github.com/search?q=repo%3AKiyotoko%2Fvault+path%3Atask_11.m&type=code):

![Search result](https://github.com/github-linguist/linguist/assets/627280/d1e6a6e1-553b-4a25-89b0-9df5dcaba612)

Unfortunately, Linguist can't do anything about the caching, but you can implement an [override](https://github.com/github-linguist/linguist/blob/master/docs/overrides.md) to force the language. You'll probably have to make a change to another file after adding the override to force a full analysis again which should then correct things.

**Reply by Kiyotoko** [reply]
Thank you very much, this has helped a lot.
